src/overlays/actors/ovl_En_Minifrog/z_en_minifrog.c

```diff
@@ -21,60 +43,602 @@ const ActorInit En_Minifrog_InitVars = {
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Minifrog_0x808A3670/func_808A44BC.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Minifrog_0x808A3670/func_808A45A8.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Minifrog_0x808A3670/func_808A4634.asm")
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_NONE,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_1,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK1,
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_NONE,
+        OCELEM_ON,
+    },
+    { 12, 14, 0, { 0, 0, 0 } },
+};
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Minifrog_0x808A3670/func_808A46E8.asm")
+static CollisionCheckInfoInit sColChkInfoInit = { 1, 12, 14, MASS_IMMOVABLE };
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Minifrog_0x808A3670/func_808A4914.asm")
+// sEyeTextures???
+static UNK_TYPE4* D_808A4D74[] = {
+    &D_060059A0,
+    &D_06005BA0,
+};
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Minifrog_0x808A3670/EnMinifrog_Update.asm")
+// gSaveContext.weekEventReg[KEY] = VALUE
+// KEY | VALUE
+static u16 isFrogReturnedFlags[] = {
+    (0x00 << 8) | 0x00, 
+    (0x20 << 8) | 0x40, 
+    (0x20 << 8) | 0x80, 
+    (0x21 << 8) | 0x01, 
+    (0x21 << 8) | 0x02,
+};
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Minifrog_0x808A3670/func_808A4AC8.asm")
+static s32 D_808A4D88 = false;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Minifrog_0x808A3670/func_808A4AF8.asm")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32_DIV1000(gravity, -800, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Minifrog_0x808A3670/func_808A4B3C.asm")
+void EnMinifrog_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnMinifrog* this = THIS;
+    int i;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 15.0f);
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_0600B538, &D_06001534, this->limbDrawTable, this->transitionDrawTable, 24);
+    CollisionCheck_SetInfo(&this->actor.colChkInfo, NULL, &sColChkInfoInit);
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+
+    if (D_808A4D88 == false) {
+        for (i = 0; i < 2; i++) {
+            D_808A4D74[i] = Lib_SegmentedToVirtual(D_808A4D74[i]);
+        }
+        D_808A4D88 = true;
+    }
+
+    this->frogIndex = (this->actor.params & 0xF);
+    if (this->frogIndex >= 5) {
+        this->frogIndex = MINIFROG_YELLOW;
+    }
+
+    this->actor.speedXZ = 0.0f;
+    this->actionFunc = EnMinifrog_Idle;
+    this->jumpState = MINIFROG_STATE_GROUND;
+    this->flags = 0;
+    this->timer = 0;
+
+    if (1) {}
+    if (!((this->actor.params & 0xF0) >> 4)) { // Frogs scattered throughout the world
```

Comment:
Needs a macro for params access.

Reply:
Done! Hopefully it's in the right style, I haven't done a macro like that before